src/blockdata/script.rs

```diff
@@ -476,50 +540,37 @@ impl Script {
 
     /// Write the assembly decoding of the script bytes to the formatter.
     pub fn bytes_to_asm_fmt(script: &[u8], f: &mut dyn fmt::Write) -> fmt::Result {
-        let mut index = 0;
-        while index < script.len() {
-            let opcode = opcodes::All::from(script[index]);
-            index += 1;
+        let mut iter = script.iter();
+        // Was at least one opcode emitted?
+        let mut at_least_one = false;
+        while let Some(byte) = iter.next() {
```

Comment:
nit: why not just `for byte in iter`?

Reply:
I tried and with `for byte in iter` the borrow checker gets angry because `iter` is mutably borrowed later on

Reply:
alright, now it is clear. @Kixunil just ignore this nit, and thank you @RCasatta for testing/clarifying

Reply:
Yeah, the iterator is used directly in the macro to advance by more bytes. Maybe worth a comment.